cl8-web/src/components/profile/ProfileTagsComponent.vue

```diff
@@ -29,52 +29,57 @@ const debug = debugLib('cl8.ProfileTagsComponent')
 
 export default {
   name: 'ProfileTagsComponent',
-    computed: {
-    list: function () {
+  computed: {
+    profileTags: function () {
       return this.$store.getters.profile.tags || []
     },
-    options: function () {
+    originalTagList: function () {
       return this.$store.getters.fullTagList
     },
-    sortedOptions: function() {
-      return sortBy(this.options, function(x){
+    sortedTagList: function() {
+      const tags = this.originalTagList
+
+      return sortBy(tags, function(x){
         return x.name ? x.name.toLowerCase() : x.toLowerCase()
       })
+      return tags
     }
   },
   data: () => {
     return {
-      input: '',      
+      input: ''
     }
 
   },
   methods: {
-    inTagList: function(tagName) {
-      if (this.list.length < 1){
+    inProfileTags: function(tagName) {
+      if (this.profileTags.length < 1){
         return false
       }
-      const tagNames = this.list.map(x => x.name)
-
+      const tagNames = this.profileTags.map(x => x.name)
       return includes(tagNames, tagName)
     },
-    toggle: function(event, val) {
-      event.preventDefault()
+    toggle: function(tag) {
+      debug('toggle', tag)
 
-      let toggleList = this.list
+      let profileTags = this.profileTags.slice()
 
-      if (this.inTagList(val.name)) {
-        remove(toggleList, function(x) { return x.name === val.name})
+      if (this.inProfileTags(tag.name)) {
+        remove(profileTags, function(x) { return x.name === tag.name})
       } else {
-        toggleList.push(val)
+        profileTags.push(tag)
       }
+      debug('profileTags', this.profileTags.map(x => x.name))
+      this.$store.commit('SET_PROFILE_TAGS', profileTags)
```

Comment:
I believe the best practice is to dispatch an action from the component that calls the mutation, but this works.

Reply:
Can you point to a style guide referring to this?

I was looking for some guidance before, and I'm aware of this

https://vuejs.org/v2/style-guide/

And this:
https://learn-vuejs.github.io/vue-patterns/useful-links/

I understand that Vuex is based around the ideas from Flux and whatnot, but I couldn't find any unambiguous, commonly used guidance here for Vuex, and it would be great to to be able to refer to, even if it's just a cheat sheet.
https://redux.js.org/style-guide/style-guide/